### PR TITLE
update go-zfs to include Solaris support

### PR DESCRIPTION
**Pull request by amitkris**
Solaris support has been merged into mistifyio/go-zfs.
This PR is to uprev go-zfs in the Docker Engine as it is a dependency for an ongoing effort towards a native port of Docker on Solaris.

### CODE REVIEW

File: hack/vendor.sh [hunk 1]
```diff
@@ -52,6 +51,10 @@ clone git github.com/miekg/dns 75e6e86cc601825c5dbcd4e0c209eab180997cd7
 clone git github.com/docker/distribution 9ec0d742d69f77caa4dd5f49ceb70c3067d39f30
 clone git github.com/vbatts/tar-split v0.9.11
 
+# get go-zfs packages
+clone git github.com/mistifyio/go-zfs 22c9b32c84eb0d0c6f4043b6e90fc94073de92fa
```
Comment (thaJeztah) [inline hack/vendor.sh]: Would it be possible for them to tag a new release?

Comment (amitkris) [inline hack/vendor.sh]: When I had pushed the changes they did not have a definitve ETA for the next release. Let me ask again.

Comment (justincormack) [inline hack/vendor.sh]: It would be nice if they had a release before docker 1.12, but I think we could merge this before that.

Comment (thaJeztah) [inline hack/vendor.sh]: Yes, I briefly discussed this with @amitkris on IRC. Having a tag would definitely be preferred, but given that we don't "control" their release cycle, using a commit is probably OK for now

**Comment by runcom** [comment]
LGTM (tag would be very welcome though)

File: hack/vendor.sh (hunk 1 above)
Comment (Mic92) [inline hack/vendor.sh]: they were pretty responsive in the past regarding that. Maybe they can release a new version, if you want.

**Comment by thaJeztah** [comment]
LGTM (we can update with a tag once it's available)

**Comment by amitkris** [comment]
thank you :) 
I am still waiting to hear back from go-zfs about the next version. As soon as they cut the next version I will file a subsequent PR to revendor it in the Engine.

**Comment by thaJeztah** [comment]
Thanks @amitkris

**Pull request merged**